airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -257,16 +251,19 @@ public SourceDiscoverSchemaRead discoverSchemaForSourceFromSourceCreate(final So
     final SourceConnection source = new SourceConnection()
         .withSourceDefinitionId(sourceCreate.getSourceDefinitionId())
         .withConfiguration(partialConfig);
-    final SynchronousResponse<AirbyteCatalog> response = synchronousSchedulerClient.createDiscoverSchemaJob(source, imageName);
+    final SynchronousResponse<String> response = synchronousSchedulerClient.createDiscoverSchemaJob(source, imageName, sourceDef.getDockerImageTag());
```

Comment:
It definitely feels like we've lost some readability with the change from `AirbyteCatalog` to `String` here. If the ID were a UUID, it'd be pretty clear, ie `SynchronousResponse<UUID>` but as a reader it feels like `String` could be anything. If this is actually a UUID, maybe we can find a way to type it as such so these response types are clear?

Reply:
I think it's a UUID, so we should be able to type it as such.

Reply:
Attempted to make this a bit clearer (on top of the UUID change indeed).